core/include/irq.h

```diff
@@ -27,6 +27,51 @@
 extern "C" {
 #endif
 
+#ifdef MODULE_CORTEXM_COMMON
```

Comment:
While I wish that we would transition to an `inline`able implementation (without `LTO=1`) by the end of next week, I doubt we will manage. So maybe it is better to have a more maintainable guard. How about this:

Every platform except CortexM defines `LEGACY_IRQ_API`. And here we check for `#ifndef LEGACY_IRQ_API`. This would also keep core platform independent (as the check is for a feature, rather than a platform).

Reply:
Ok I did that defining them in `cpu_conf.h`, le me know if you think there is a better place.

Reply:
I would also have put it just there